site/docs/hosting/vps.md

```diff
@@ -45,131 +120,486 @@ cd /etc/systemd/system
 
-```sh
-systemctl status <service_name>
+PM2 offers two ways to create an application:
+
+1. Use the command line interface.
+2. Use the [configuration file](https://pm2.keymetrics.io/docs/usage/application-declaration/).
+
+The first method is convenient when getting to know PM2.
+However, during deployment, you should use the second method, which is what we did in our case.
+
+On our server, in the directory where the bot assembly is stored, there is a file `ecosystem.config.js` with the following contents:
+
+```js
+module.exports = {
+  apps: [{
+    name: "<app-name>",
+    script: "<start-command>",
+  }],
+};
 ```
 
-#### Restart
+where `<app-name>` can be any identifier, and `<start command>` must be the actual command to start the bot, for example, `node bot.js` or `npm run start`, etc.
 
-```sh
-systemctl restart <service_name>
+#### Managing the Application
+
+Below are the commands you can use to control the application.
+
+```sh:no-line-numbers
+# If the `ecosystem.config.js` file is in the current directory,
+# you can specify nothing to start the application.
+# If the application is already running, this command will restart it.
+pm2 start
+
+# All of the following commands require you to specify the name of the application
+# or the `ecosystem.config.js` file.
+# To apply the action to all applications, specify `all`.
+
+# To restart the app
+pm2 restart <<app-name>>
+
+# To reload the app
+pm2 reload <<app-name>>
+
+# To stop the app
+pm2 stop <<app-name>>
+
+# To delete an app
+pm2 delete <<app-name>>
 ```
 
-#### Stop
+#### Saving the Application Operation
 
-```sh
-systemctl stop <service_name>
+If the server reboots, your bot will not resume working.
+For the bot to resume work, you need to prepare PM2 for this.
+
+On the server in the terminal, run the following command:
+
+```sh:no-line-numbers
+pm2 startup
 ```
 
-#### Don't Run on Boot
+You will be provided with a command that you must execute to make PM2 automatically start after the server reboots.
 
-```sh
-systemctl disable <service_name>
+Then run one more command:
+
+```sh:no-line-numbers
+pm2 save
 ```
 
-## PM2 (Node.js only)
+This command will save the list of current applications so that they can be launched after a server reboot.
 
-PM2 is a daemon process manager for Node.js that will help you manage and keep your app online 24/7.
+If you have created a new application and want to save it as well, simply run `pm2 save` again.
 
-### Installing
+## Running the Bot on Webhooks
 
-```sh
-npm install pm2@latest -g
+To run a bot on webhooks, you will need to use a web framework and not call `bot.start()`.
 
-# If using Yarn
-yarn global add pm2
+Here's a sample code to start the bot using `fastify`, which you should add to the bot's main file:
+
+```ts
+import { webhookCallback } from "grammy";
+import { fastify } from "fastify";
+
+const server = fastify();
+
+server.post(`/${bot.token}`, webhookCallback(bot, "fastify"));
+
+server.listen();
 ```
```

Comment:
Do you have an example of setting up a webhook using `node:http`? I think it would be better to use native modules instead of dependencies, but I'm too lazy to figure it out myself. I'll take the example for Deno from [here](https://grammy.dev/hosting/deno-deploy.html#preparing-your-code)